### DISCUSSION TITLE
Include headers frome another Fprime module

### DISCUSSION THREAD

**Discussion by SMorettini** [nasa/fprime#2464] [Q&A] [answered]
I have my component `B` depending on `A`. Inside `B` I include one header file from `A`. In Fprime 3.3.1, to obtain that was enough to add `A` in `MOD_DEPS` of `B`, in Fprime 3.4.1 it doesn't work anymore because the header cannot be found. It seems to me that the change causing this behaviour is the following line:
https://github.com/nasa/fprime/blob/8a1612c0dd1ec18ada8162d92d37ff7317835129/cmake/target/build.cmake#L66

1. Why it was commented? 
2. What is the correct way to make the headers in `A` available for `B`?
3. Should I use `target_include_directory` on any component I want to make the header available?

**Comment by LeStarch** [comment, marked as answer]
This was commented out because F´ handwritten code should use full paths from the project, framework, or library root to avoid ambiguity between components with the same name.

Instead of doing `#include <A.hpp>` use instead `#include <Folder/Folder/A.hpp>` and it will work. Here is an example:

```c++
// Risky code:
#include <CommandDispatcherImpl.hpp>

// Better code:
#include <Svc/CommandDispatcherImpl.hpp>
```

This was always the intended way to deal with header includes in F´ and is the expected style/standard.  At some point that above line of code leaked in, and we removed it to once again enforce standards.

You may, of course, add that line back in within your CMake system....however it is strongly discouraged as it makes it harder to share code to others, and borrow code from others.

Please let me know if you have further trouble.

**Reply by LeStarch** [reply]
Additionally, we recommend that a component `B` never depend on a component `A` directly.  Instead break out common code into a module `C` that both `B` and `A` depend on.  This is why our ports are separate modules from the components that use them.

The reason is that it keeps components separated for things like unit tests where you only want a single component linked-in.
